qubes/storage/callback.py

```diff
@@ -442,7 +442,7 @@ def _assert_initialized(self, **kwargs):
     def _callback(self, cb, cb_args=None, **kwargs):
         if cb_args is None:
             cb_args = []
-        vol_args = [self.name, self.vid, *cb_args]
+        vol_args = [self.name, self.vid, self.source, *cb_args]
```

Comment:
And this is yet another example why `filter(None, ...)` on args is a terrible idea. If `self.source` is None for a given volume, then other arguments (like `size` for resize for example) will be in different place.

Reply:
Fully true.

But I had (admittedly somewhat silently) dropped the line in https://github.com/QubesOS/qubes-core-admin/commit/4becf29c342b4ddd8afd6425608054a145d1c5db#diff-271f0ec3445d2fb476cb711090e750b8L252 anyway.

The other `filter()` two lines below is still somewhat required to remove an entirely empty args String `''` / avoid adding an unnecessary space after a command, if there are no arguments (args is a String by then).